symbolic-ppdb/src/format/mod.rs

```diff
@@ -276,6 +284,20 @@ impl<'data> PortablePdb<'data> {
                 _ => return Err(FormatErrorKind::UnknownStream.into()),
             }
         }
+
+        // Read source link mappings.
+        // https://github.com/dotnet/runtime/blob/main/docs/design/specs/PortablePdb-Metadata.md#source-link-c-and-vb-compilers
+        const SOURCE_LINK_KIND: Uuid = uuid::uuid!("CC110556-A091-4D38-9FEC-25AB9A351A6A");
+        for cdi in CustomDebugInformationIterator::new(&result, SOURCE_LINK_KIND)? {
+            let cdi = cdi?;
+            // Note: only handle module #1 (do we actually handle multiple modules in any way??)
+            if let (metadata::CustomDebugInformationTag::Module, 1) = (cdi.tag, cdi.value) {
+                let json = String::from_utf8_lossy(result.get_blob(cdi.blob)?);
+                result.source_link_mappings.add_mappings(&json)?;
+            }
+        }
+        result.source_link_mappings.sort();
```

Comment:
I would prefer it if this happened in a constructor function on `SourceLinkMappings`. Then `SourceLinkMappings` could always be sorted by construction and there would be no need to expose that detail outside of the `sourcelinks` module.

Reply:
Ok, refactored it to a single constructor